packaging/ubuntu-14.04/snapd.postrm

```diff
@@ -22,6 +22,11 @@ if [ "$1" = "purge" ]; then
     # snap.mount.service is a trusty thing
     systemctl_stop snap.mount.service
 
+    # undo any bind mounts to /var/snap done by parallel installs
+    if grep -q " /var/snap /var/snap" /proc/self/mountinfo; then
```

Comment:
This will also unmount /var/snap that is legitimately mounted from another media but not using parallel installs for classic thing.

We could not do this if we always had a mount unit and let systemd handle things but this does leave the question open of how to configure /var/snap to be on another partition/disk correctly.

Reply:
It's not impossible, it's just less reliable than I wished we can offer. For things in /run it's unlikely this will ever be a problem. For `/var/snap` I can see people mount things there. What will happen if they just format an empty drive, mkdir /var/snap on the partition and mount that?

Having said that I think the problem is deeper. It is the fundamental incompatibility between /var/snap's special propagation as seen on the host and the fact people may want to set it up as a mount point for stuff.

Perhaps we should not bind mount `/var/snap` over itself if it is a mount point and merely change the settings of an existing mount point? With some more elaborate code we could then check how the mount point looks like and chose to leave it as-is.

Reply:
IMHO it seems like the situation where this is a problem is rare, so I think it would be good to get this in and either handle as a follow-up or fix that corner case when we know it's a problem.